### PR TITLE
Provide Microsoft.Extensions.Caching.Abstractions package readme

### PR DESCRIPTION
**Pull request by eNeRGy164**
Fill out details for the `Microsoft.Extensions.Caching.Abstractions` NuGet package readme.

See #92228

### CODE REVIEW

**Comment by ViktorHofer** [comment]
@dotnet/area-extensions-options please review this PR today if possible so that we can still get this change backported into .NET 8.

**Comment by tarekgh** [comment]
tagging the owners @dotnet/area-extensions-caching

**Comment by ViktorHofer** [comment]
oops, sorry for the wrong ping.

File: src/libraries/Microsoft.Extensions.Caching.Abstractions/src/PACKAGE.md [hunk 1]
```diff
@@ -2,43 +2,51 @@
 
 <!-- A description of the package and where one can find more documentation -->
 
+Microsoft.Extensions.Caching.Abstractions offers the basics to create and use caching in your applications.
 
+This library doesn’t hold the actual caches, but it defines how caches should behave.
```
Comment (Jozkee) [inline src/libraries/Microsoft.Extensions.Caching.Abstractions/src/PACKAGE.md:7]:
```suggestion
This library defines how in-memory and distributed caches should be implemented; it doesn’t contain any cache implementation.
```

File: src/libraries/Microsoft.Extensions.Caching.Abstractions/src/PACKAGE.md [hunk 2]
```diff
@@ -2,43 +2,51 @@
...
+With this, you or other libraries can build various types of caches that all apps can use in the same way, whether the data is kept in memory, in files, or even across a network.
```
Comment (Jozkee) [inline src/libraries/Microsoft.Extensions.Caching.Abstractions/src/PACKAGE.md:8]:
```suggestion
With the abstractions provided in this library, various types of caches can be built and used interchangeably, whether the data is kept in memory, in files, or even across a network.
```

File: src/libraries/Microsoft.Extensions.Caching.Abstractions/src/PACKAGE.md [hunk 3]
```diff
@@ -2,43 +2,51 @@
...
 
 ## Key Features
 
 <!-- The key features of this package -->
 
-*
-*
-*
+* Basic rules and tools for building different kinds of caches.
```
Comment (Jozkee) [inline src/libraries/Microsoft.Extensions.Caching.Abstractions/src/PACKAGE.md:14]:
```suggestion
* Interfaces for building and using in-memory and distributed caches.
```

File: src/libraries/Microsoft.Extensions.Caching.Abstractions/src/PACKAGE.md [hunk 4]
```diff
@@ -2,43 +2,51 @@
...
+* Makes it easier to create or use caches in your apps, whether they're simple or complex.
```
Comment (Jozkee) [inline src/libraries/Microsoft.Extensions.Caching.Abstractions/src/PACKAGE.md:15]: This sounds vague to me. How does it make it easier? What's simple and what's complex? Consider removing it.
```suggestion
```

**Review by jozkee — APPROVED** [review]

**Comment by ViktorHofer** [comment]
Thanks a lot 👍

**Pull request merged**